### ISSUE TITLE
Deleted node won't show as children on restore

### ISSUE THREAD

**Issue by nicolaeschbach** [cyber-geiger/toolbox-storage#28]
**Describe the bug**
When a node is removed and added again, it is no longer a tombstone, but it is also no longer a child of the parent node. Is this a bug or intentional?

**Environmental information**
geiger_localstorage: ^0.6.15

**To Reproduce**
```dart
import 'package:geiger_localstorage/geiger_localstorage.dart';
import 'package:test/expect.dart';
import 'package:test/scaffolding.dart';


void main() async {
  test('Test GetChildren', () async {
    StorageController controller = GenericController("testOwner", DummyMapper("testOwner"));
    Node node1 = NodeImpl("Node1","testOwner", ":Enterprise");
    await controller.add(node1);

    await Future.delayed(Duration(milliseconds: 50));
    expect((await controller.getNodeOrTombstone(":Enterprise:Node1")).tombstone, isFalse);

    await controller.delete(":Enterprise:Node1");
    expect((await controller.getNodeOrTombstone(":Enterprise:Node1")).tombstone, isTrue);

    await controller.addOrUpdate(node1);
    expect((await controller.getNodeOrTombstone(":Enterprise:Node1")).tombstone, isFalse);
    expect((await (await controller.get(":Enterprise")).getChildren()).values.toList().isNotEmpty, isTrue);
  });
}
```

**Expected behavior**
A node should be a children again, if restored by addOrUpdate().

**Comment by martingwerder** [comment]
May I ask why you added the line ```await Future.delayed(Duration(milliseconds: 50));```? It seems sensless to me but maybe there is a second problem lurkiung there. Can you elaborate?

**Comment by nicolaeschbach** [comment]
Yes, you are right. This Future.delayed statements may be removed. The issue appears anyhow, and it's not to to avoid any other issues.

**Comment by martingwerder** [comment]
A fix is published can you confirm that it is working?

**Comment by nicolaeschbach** [comment]
I can not see any fixes so far. Do you know when the latest version will be released on pub.dev?

**Comment by martingwerder** [comment]
Pushed now to pub.dev

 

Von: nicolaeschbach ***@***.***> 
Gesendet: Montag, 13. Dezember 2021 07:21
An: cyber-geiger/toolbox-storage ***@***.***>
Cc: Martin Gwerder ***@***.***>; Comment
***@***.***>
Betreff: Re: [cyber-geiger/toolbox-storage] Deleted node won't show as
children on restore (Issue #28)

 

I can not see any fixes so far. Do you know when the latest version will be
released on pub.dev?

-
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.co
m%2Fcyber-geiger%2Ftoolbox-storage%2Fissues%2F28%23issuecomment-992145404&da
ta=04%7C01%7Cmartin.gwerder%40fhnw.ch%7C3719aa0bedf7444a7be008d9be00eb29%7C9
d1a5fc8321e4101ae63530730711ac2%7C0%7C0%7C637749733419971581%7CUnknown%7CTWF
pbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%
7C3000&sdata=iOOtNSY6ZgzkO8AY9iIYHNHhgmXQ06dh4xG1ZqMFFsg%3D&reserved=0> , or
unsubscribe
<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.co
m%2Fnotifications%2Funsubscribe-auth%2FAPXCJIFTVAK5WT5ADBBIHW3UQWGGRANCNFSM5
JJZTCTQ&data=04%7C01%7Cmartin.gwerder%40fhnw.ch%7C3719aa0bedf7444a7be008d9be
00eb29%7C9d1a5fc8321e4101ae63530730711ac2%7C0%7C0%7C637749733419971581%7CUnk
nown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXV
CI6Mn0%3D%7C3000&sdata=QCr9QVwM3LaBj7DmbQJXUh8kwO%2FqTlH8EC%2BtsqPVc4o%3D&re
served=0> .
Triage notifications on the go with GitHub Mobile for iOS
<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.appl
e.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%
26pt%3D524675&data=04%7C01%7Cmartin.gwerder%40fhnw.ch%7C3719aa0bedf7444a7be0
08d9be00eb29%7C9d1a5fc8321e4101ae63530730711ac2%7C0%7C0%7C637749733419971581
%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWw
iLCJXVCI6Mn0%3D%7C3000&sdata=1vdIquP7lyM52Wpg79NFKhufy5OZtTWTjs0VVR6MOME%3D&
reserved=0>  or Android.
<https://github.com/notifications/beacon/APXCJIEJTWRDTWX6AVG7JWTUQWGGRA5CNFS
M5JJZTCT2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOHMRO77A
.gif>

**Comment by nicolaeschbach** [comment]
Confirm. This issue is fixed with the 0.6.17 release.
